elastic/gobench
# Parse apmbench results, add ES basicauth

## Description
Updates the benchmark parsing to allow to optionally parse the extra
metrics that are published by apmbench. Modified the Elasticsearch
request handling slightly to allow us to authenticate using basicauth
and also work with versions of the stack `8.0.0+`.

Last, created a Go module to allow the project to work with newer Go
versions.

## How to test this changes

1. Start a local Elastic stack.
2. `cat testdata/benchmark-result.txt| go run main.go -es http://localhost:9200 -es-username admin -es-password changeme -tag "apm-server_version=8.1.2,size=4096`.
3. Verify it is indexed in the `gobench` index with the right mappings.

## Related issues

Part or elastic/apm-server#7540
